src/main/java/org/peergos/EmbeddedIpfs.java

```diff
@@ -0,0 +1,157 @@
+        records.close();
+        return node.stop();
+    }
+
+    public static Blockstore buildBlockStore(Config config, Path ipfsPath) {
+        Path blocksPath = ipfsPath.resolve("blocks");
+        File blocksDirectory = blocksPath.toFile();
+        if (!blocksDirectory.exists()) {
+            if (!blocksDirectory.mkdir()) {
+                throw new IllegalStateException("Unable to make blocks directory");
+            }
+        } else if (blocksDirectory.isFile()) {
+            throw new IllegalStateException("Unable to create blocks directory");
+        }
+        FileBlockstore blocks = new FileBlockstore(blocksPath);
+        Blockstore blockStore;
+        if (config.datastore.filter.type == FilterType.BLOOM) {
+            blockStore = FilteredBlockstore.bloomBased(blocks, config.datastore.filter.falsePositiveRate);
+        } else if(config.datastore.filter.type == FilterType.INFINI) {
+            blockStore = FilteredBlockstore.infiniBased(blocks, config.datastore.filter.falsePositiveRate);
+        } else if(config.datastore.filter.type == FilterType.NONE) {
+            blockStore = blocks;
+        } else {
+            throw new IllegalStateException("Unhandled filter type: " + config.datastore.filter.type);
+        }
+        return config.datastore.allowedCodecs.codecs.isEmpty() ?
+                blockStore : new TypeLimitedBlockstore(blockStore, config.datastore.allowedCodecs.codecs);
+    }
+
+    public static EmbeddedIpfs build(Path ipfsPath,
+                                     Blockstore blocks,
+                                     List<MultiAddress> swarmAddresses,
+                                     List<MultiAddress> bootstrap,
+                                     IdentitySection identity,
+                                     Optional<HttpProtocol.HttpRequestProcessor> handler) {
+        ProvidingBlockstore blockstore = new ProvidingBlockstore(blocks);
+        Path datastorePath = ipfsPath.resolve("datastore").resolve("h2.datastore");
+        DatabaseRecordStore records = new DatabaseRecordStore(datastorePath.toString());
+        ProviderStore providers = new RamProviderStore();
+
+        int hostPort = swarmAddresses.get(0).getPort();
+        HostBuilder builder = new HostBuilder().setIdentity(identity.privKeyProtobuf).listenLocalhost(hostPort);
+        if (! builder.getPeerId().equals(identity.peerId)) {
+            throw new IllegalStateException("PeerId invalid");
+        }
+        Multihash ourPeerId = Multihash.deserialize(builder.getPeerId().getBytes());
+
+        Kademlia dht = new Kademlia(new KademliaEngine(ourPeerId, providers, records), false);
+        CircuitStopProtocol.Binding stop = new CircuitStopProtocol.Binding();
+        CircuitHopProtocol.RelayManager relayManager = CircuitHopProtocol.RelayManager.limitTo(builder.getPrivateKey(), ourPeerId, 5);
+        BlockRequestAuthoriser authoriser = (c, b, p, a) -> CompletableFuture.completedFuture(true);
```

Comment:
can authoriser be abstracted out, so it can be configurable/swapable?

Reply:
Done